contrib/pkg/awsprivatelink/enable.go

```diff
@@ -0,0 +1,219 @@
+2) Create a Secret with the above credential.
+3) Add a reference to the Secret in HiveConfig.spec.awsPrivateLink.credentialsSecretRef.
+4) Add the active cluster's VPC to the list of HiveConfig.spec.awsPrivateLink.associatedVPCs`,
+		Run: func(cmd *cobra.Command, args []string) {
+			if err := opt.Complete(cmd, args); err != nil {
+				return
+			}
+			if err := opt.Validate(cmd, args); err != nil {
+				return
+			}
+			if err := opt.Run(cmd, args); err != nil {
+				return
+			}
+		},
+	}
+
+	flags := cmd.Flags()
+	flags.StringVar(&opt.dnsRecordType, "dns-record-type", "Alias", "HiveConfig.spec.awsPrivateLink.dnsRecordType")
+	return cmd
+}
+
+func (o *enableOptions) Complete(cmd *cobra.Command, args []string) error {
+	// Get controller-runtime dynamic client
+	if err := configv1.Install(scheme.Scheme); err != nil {
+		log.WithError(err).Fatal("Failed to add Openshift configv1 types to the default scheme")
+	}
+	dynamicClient, err := hiveutils.GetClient()
+	if err != nil {
+		log.WithError(err).Fatal("Failed to create controller-runtime client")
+	}
+	o.dynamicClient = dynamicClient
+
+	// Get current user's home directory
+	o.homeDir = "."
+	u, err := user.Current()
+	if err != nil {
+		log.WithError(err).Fatal("Failed to get the current user")
+	}
+	o.homeDir = u.HomeDir
+
+	// Get AWS clients
+	o.region, o.infraId, err = o.getRegionAndInfraId()
+	if err != nil {
+		log.WithError(err).Fatal("Failed to get region and infraID from Infrastructure/cluster")
+	}
+	log.Debugf("Found region = %v, infraId = %v", o.region, o.infraId)
+	awsClients, err := awsclient.NewClientFromSecret(nil, o.region)
+	if err != nil {
+		log.WithError(err).Fatal("Failed to create AWS clients")
+	}
+	o.awsClients = awsClients
+
+	return nil
+}
+
+func (o *enableOptions) Validate(cmd *cobra.Command, args []string) error {
+	switch o.dnsRecordType {
+	case "Alias", "ARecord":
+	default:
+		log.Fatal(`--dns-record-type must be one of "Alias", "ARecord"`)
+	}
+
+	return nil
+}
+
+func (o *enableOptions) Run(cmd *cobra.Command, args []string) error {
+	// Get HiveConfig
+	hiveConfig := &hivev1.HiveConfig{}
+	if err := o.dynamicClient.Get(context.Background(), types.NamespacedName{Name: "hive"}, hiveConfig); err != nil {
+		log.WithError(err).Fatal("Failed to get HiveConfig/hive")
+	}
+	if hiveConfig.Spec.AWSPrivateLink != nil {
+		log.Fatal("AWS Private Link is already enabled")
+	}
+
+	// Get active cluster's VPC, filtering by infra-id
+	describeVPCsOutput, err := o.awsClients.DescribeVpcs(&ec2.DescribeVpcsInput{
+		Filters: []*ec2.Filter{
+			{
+				Name:   aws.String("tag:Name"),
+				Values: []*string{aws.String(o.infraId + "-vpc")},
+			},
+		},
+	})
+	if err != nil {
+		log.WithError(err).Fatal("Failed to get VPC of the active cluster")
+	}
+	if len(describeVPCsOutput.Vpcs) == 0 {
+		log.Fatal("VPC of the active cluster not found")
+	}
+	vpcID := *describeVPCsOutput.Vpcs[0].VpcId
```

Comment:
It should be impossible for there to be more than one entry, right?

Reply:
I just switched to filtering using the "kubernetes.io/cluster/INFRA-ID" key for consistency (that's mostly what I'm doing elsewhere). 

Regardless of the method of filtering, technically there could be multiple entries, as one may tag a random VPC with the same key (or give another VPC the same name, which is allowed even in the same region). That said, this is highly unlikely as I could not think of any reason for somebody to actually do this.

Reply:
Yep. I added an IF statement s.t. the executable errors out in case multiple VPCs are found.